server/context.go

```diff
@@ -83,6 +83,9 @@ type Context struct {
 	// communicated between hosts on the gossip network.
 	GossipInterval time.Duration
 
+	// // Enables running the node as a single-node in-memory cluster.
+	TransientSingleNode bool
```

Comment:
It's weird that this is in the `server` package; I would have thought that the `cli` package could handle this entirely from the outside (i.e. as soon as the flags are parsed, `if *transientSIngleNode { ctx.GossipBootstrap = "self=" }`)